composer/trainer/trainer.py

```diff
@@ -1320,7 +1320,7 @@ def __init__(
         if self._train_data_spec is not None:
             self.state.set_dataloader(self._train_data_spec.dataloader, train_dataloader_label,
                                       train_subset_num_batches)
-            if isinstance(self.state.device, DeviceTPU):
+            if self.state.device.dist_backend == 'xla':
```

Comment:
```suggestion
            if isinstance(self.state.device, DeviceNeuron):
```

Reply:
I'll like to keep this as self.state.device.dist_backend so it works for all xla based devices (rather than having to add more devices to the isinstance check if another one comes along), what do you think?

Reply:
Ah got it, makes sense! Thanks